app_dart/lib/src/request_handlers/push_build_status_to_github.dart

```diff
@@ -6,122 +6,140 @@ import 'dart:async';
-          } catch (error) {
-            log.error('Failed to post status update to ${slug.fullName}#${pr.number}: $error');
-          }
+    scheduler.setLogger(log);
+
+    final String repo = request.uri.queryParameters[fullNameRepoParam] ?? 'flutter/flutter';
+    final RepositorySlug slug = RepositorySlug.full(repo);
+
+    for (String branch in await config.getSupportedBranches(slug)) {
+      final Commit tipOfTreeCommit = Commit(sha: branch, repository: slug.fullName);
+      final SchedulerConfig schedulerConfig = await scheduler.getSchedulerConfig(tipOfTreeCommit);
+      final List<LuciBuilder> postsubmitBuilders =
+          await scheduler.getPostSubmitBuilders(tipOfTreeCommit, schedulerConfig);
+      final LuciService luciService = luciServiceProvider(this);
+      final Map<LuciBuilder, List<LuciTask>> luciTasks = await luciService.getRecentTasks(builders: postsubmitBuilders);
+
+      String status = GithubBuildStatusUpdate.statusSuccess;
+      for (List<LuciTask> tasks in luciTasks.values) {
+        final String latestStatus = await _getLatestStatus(tasks);
+        if (status == GithubBuildStatusUpdate.statusSuccess && latestStatus == GithubBuildStatusUpdate.statusFailure) {
+          status = GithubBuildStatusUpdate.statusFailure;
         }
       }
-
-      /// Whenever github status is updated, [update.updates] will be synchronized in
-      /// datastore [GithubBuildStatusUpdate].
-      await datastore.insert(updates);
-      log.debug('Committed all updates');
+      await _insertBigquery(slug, status, branch, log, config);
+      final DatastoreService datastore = datastoreProvider(config.db);
+      await _updatePRs(slug, status, datastore);
+      log.debug('All the PRs for $repo have been updated with $status');
     }
 
     return Body.empty;
   }
 
-  Future<void> _insertBigquery(BuildStatus buildStatus, String branch) async {
-    // Define const variables for [BigQuery] operations.
-    const String projectId = 'flutter-dashboard';
-    const String dataset = 'cocoon';
-    const String table = 'BuildStatus';
-
-    final TabledataResourceApi tabledataResourceApi = await config.createTabledataResourceApi();
-    final List<Map<String, Object>> requestRows = <Map<String, Object>>[];
-
-    requestRows.add(<String, Object>{
-      'json': <String, Object>{
-        'Timestamp': DateTime.now().millisecondsSinceEpoch,
-        'Status': buildStatus.value,
-        'Branch': branch,
-      },
-    });
-
-    // Obtain [rows] to be inserted to [BigQuery].
-    final TableDataInsertAllRequest request = TableDataInsertAllRequest.fromJson(<String, Object>{'rows': requestRows});
+  Future<void> _updatePRs(RepositorySlug slug, String status, DatastoreService datastore) async {
+    final GitHub github = await config.createGitHubClient(slug);
+    final List<GithubBuildStatusUpdate> updates = <GithubBuildStatusUpdate>[];
+    await for (PullRequest pr in github.pullRequests.list(slug)) {
```

Comment:
This will list PRs of master branch only, and will repeat same updates within the branch loop.